### ISSUE TITLE
normalization does not generate deterministic ids

### ISSUE THREAD

**Issue by ekg** [ekg/vg#117]
A wish, as it would make testing easier. If we sort the graph the sort should stably-sort nodes that have the same rank (using sequence, for instance) such that when we assign ids to them the node with the same sequence will always get the same id.

**Comment by adamnovak** [comment]
Do we still want to make the topological sort completely deterministic?

**Comment by faithokamoto** [comment]
Yeah, I don't think there's really any way to guarantee a stable sort regardless of the input node IDs. Even if we broke ties on sequence, it's possible to have nodes with the same rank and sequence (e.g. see original graph in #4710) Our tests work fine now and there are ways to work around this when needed; if someone really wants this, please reopen and/or make a new issue to explain a specific use case.
